internal/batches/service/remote.go

```diff
@@ -98,13 +113,122 @@ func (svc *Service) CreateBatchSpecFromRaw(
 		"allowIgnored":     allowIgnored,
 		"allowUnsupported": allowUnsupported,
 		"noCache":          noCache,
+		"batchChange":      batchChange,
 	}).Do(ctx, &resp); err != nil || !ok {
 		return "", err
 	}
 
 	return resp.CreateBatchSpecFromRaw.ID, nil
 }
 
+// UploadMounts uploads file mounts to the server.
+func (svc *Service) UploadMounts(workingDir string, batchSpecID string, steps []batches.Step) error {
+	if err := svc.areServerSideBatchChangesSupported(); err != nil {
+		return err
+	}
+
+	body := &bytes.Buffer{}
+	w := multipart.NewWriter(body)
+
+	var count int
+	for _, step := range steps {
+		for _, mount := range step.Mount {
+			total, err := handlePath(w, workingDir, mount.Path, count)
+			if err != nil {
+				return err
+			}
+			count += total
+		}
+	}
+
+	if err := w.WriteField("count", strconv.Itoa(count)); err != nil {
```

Comment:
Can't count be derived from the payload? What is is required for here?

Reply:
This is the number of files being uploaded. It could be derived if I look at the `file_%d` field. But this would require to parse this (and potentially look at every field since order isn't guaranteed).

Reply:
Yea, with this current implementation. With this bulk upload, it needs to know how many files are in the request (so it can lookup the files from the payload)